src/bugsplat/bugsplat.py

```diff
@@ -1,100 +1,120 @@
+    setDefaultEmail = set_default_email
+    setDefaultUser = set_default_user
+
+    def post(self,
+             ex: BaseException or str,
+             additional_file_paths: List[PathLike] = None,
+             app_key: str = '',
+             description: str = '',
+             email: str = '',
+             user: str = ''):
+
+        if not additional_file_paths:
+            additional_file_paths = self.additional_file_paths
+
+        self.logger.warning('\nBugSplat caught an Unhandled Exception!\n')
 
         # TODO BG what if ex is not defined? Do we care?
-        # https://stackoverflow.com/questions/3702675/how-to-catch-and-print-the-full-exception-traceback-without-halting-exiting-the
-        callstack = self._convertExceptionToJson(ex)
-        exceptionMessage = str(ex)
+        # https://stackoverflow.com/questions/3702675/how-to-catch-and-print-the-full
+        # -exception-traceback-without-halting-exiting-the
+        callstack = self._convert_exception_to_json(ex)
 
-        print('About to post crash to database ' + self.database + '...')
-        print('\n')
+        self.logger.warning(f'About to post crash to database {self.database}...\n')
 
-        url = 'https://' + self.database + '.bugsplat.com/post/py/'
-        if (not appKey): appKey = self.appKey
-        if (not description): description = self.description
-        if (not email): email = self.email
-        if (not user): user = self.user
-        if (len(additionalFilePaths) == 0): additionalFilePaths = self.additionalFilePaths
-        files = self._createFilesForPost(additionalFilePaths)
+        url = f'https://{self.database}.bugsplat.com/post/py/'
+        files = self._create_files_for_post(additional_file_paths)
 
         data = {
             'database': self.database,
             'appName': self.application,
             'appVersion': self.version,
-            'appKey': appKey,
-            'description': description,
-            'exceptionMessage': exceptionMessage,
-            'email': email,
-            'user': user,
+            'appKey': app_key or self.app_key,
+            'description': description or self.description,
+            'exceptionMessage': str(ex),
+            'email': email or self.email,
+            'user': user or self.user,
             'callstack': callstack
         }
 
         try:
-            response = requests.post(url, files = files, data = data)
+            response = requests.post(url, files=files, data=data)
 
-            if (response.status_code !=  200):
-                raise Exception('Status: ' + str(response.status_code) + '\n' + 'Message: ' + response.text)
+            if response.status_code != 200:
+                raise Exception(
+                    f'Status: {response.status_code} \n Message: {response.text}'
+                )
 
-            print('Crash posted successfully!')
+            self.logger.warning('Crash posted successfully!')
```

Comment:
Yeah you're probably right, semantically warning doesn't make much sense. I really only set it to warning because otherwise the default logger doesn't print info messages to console. I would just have to add some logger configuration, which is probably a good idea anyways.